papaparse.js

```diff
@@ -852,7 +852,18 @@ License: MIT
 		{
 			try
 			{
-				queue.push(typeof chunk === 'string' ? chunk : chunk.toString(this._config.encoding));
+				var encoding = this._config.encoding;
+				var decoder = this._config.decoder;
+				var decodedString = '';
+
+				if (typeof decoder === 'function') {
+					decodedString = decoder(chunk, encoding);
+				} else if (typeof chunk === 'string') {
```

Comment:
For me it is strange to use an if/else with different variables. 

What if decoder is not a function?